fbw-a380x/src/wasm/systems/a380_systems/src/fuel/fuel_quantity_management_system.rs

```diff
@@ -0,0 +1,614 @@
+            } else {
+                fuel_system.set_tank_quantity(tank as usize, current_quantity + sign * delta);
+            }
+            remaining_delta -= delta;
+        }
+        true
+    }
+
+    fn execute_instant_refuel(
+        &mut self,
+        fuel_system: &mut FuelSystem<11>,
+        refuel_panel_input: &mut IntegratedRefuelPanel,
+        desired_quantities: HashMap<A380FuelTankType, Mass>,
+    ) {
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::LeftOuter as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::LeftOuter)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::RightOuter as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::RightOuter)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::LeftMid as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::LeftMid)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::RightMid as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::RightMid)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::LeftInner as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::LeftInner)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::RightInner as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::RightInner)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::FeedOne as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::FeedOne)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::FeedFour as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::FeedFour)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::FeedTwo as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::FeedTwo)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::FeedThree as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::FeedThree)
+                .unwrap_or(&Mass::default()),
+        );
+        fuel_system.set_tank_quantity(
+            A380FuelTankType::Trim as usize,
+            *desired_quantities
+                .get(&A380FuelTankType::Trim)
+                .unwrap_or(&Mass::default()),
+        );
```

Comment:
You already have a function for it:
```suggestion
        for tank_type in A380FuelTankType::iterator() {
            fuel_system.set_tank_quantity(
                tank_type as usize,
                *desired_quantities
                    .get(&tank_type)
                    .unwrap_or(&Mass::default()),
            );
        }
```